### PR TITLE
Added PlayerDeepSleepEvent

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: Spigot-Server-Patches/0704-Added-PlayerSleepEvent.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+diff --git a/src/main/java/net/minecraft/world/entity/player/EntityHuman.java b/src/main/java/net/minecraft/world/entity/player/EntityHuman.java
+index ec0956a98c133bcd3d4f92f696c667eab6ff98f1..d410cfa49c40887487de7a9b9e3f99fd8086b3f5 100644
+--- a/src/main/java/net/minecraft/world/entity/player/EntityHuman.java
++++ b/src/main/java/net/minecraft/world/entity/player/EntityHuman.java
+@@ -235,6 +235,9 @@ public abstract class EntityHuman extends EntityLiving {
+ 
+         if (this.isSleeping()) {
+             ++this.sleepTicks;
++            if (this.sleepTicks == 100) {
```
Comment (kennytv) [inline Spigot-Server-Patches/0704-Added-PlayerSleepEvent.patch]: Paper comments missing

File: Spigot-API-Patches/0285-Added-PlayerSleepEvent.patch [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a player has slept long enough
++ * to count as passing the night/storm.
++ * <p>
++ * Cancelling this event will prevent the player from being counted as sleeping
```
Comment (kennytv) [inline Spigot-API-Patches/0285-Added-PlayerSleepEvent.patch]:
```suggestion
+ * Cancelling this event will prevent the player from being counted as deeply sleeping
```

**Review by kennytv — CHANGES_REQUESTED** [review]
I would name the event `PlayerDeepSleepEvent` (or PlayerEnterDeepSleepEvent, or similar), since imo it would otherwise (intuitively) be confused with entering the sleeping pose initially

File: Spigot-Server-Patches/0704-Added-PlayerSleepEvent.patch (hunk 1 above)
Comment (Machine-Maker) [reply Spigot-Server-Patches/0704-Added-PlayerSleepEvent.patch]: Done

**Review by MiniDigger — APPROVED** [review]

**Review by kennytv — APPROVED** [review]

**Pull request merged**